src/org/zaproxy/zap/extension/soap/SOAPActionSpoofingActiveScanner.java

```diff
@@ -113,6 +113,14 @@ public void scan() {
 				String[] soapActions = ImportWSDL.getInstance().getSourceSoapActions(originalMsg);
 
 				boolean endScan = false;
+				if (soapActions == null || soapActions.length == 0 ) {
+					// No actions to spoof
+					if (log.isDebugEnabled()) {
+						log.debug("Skipping " + getName() + " because no actions were found.");
+					}
+					getParent().pluginSkipped(getId(), Constant.messages.getString("soap.soapactionspoofing.skip.reason"));
```

Comment:
Should just return? It could miss actions in other messages if skipped.

Reply:
Fair enough, change inbound.